drivers/usb_c/tcpc/rt1715.c

```diff
@@ -0,0 +1,712 @@
+	.transmit_data = rt1715_tcpc_transmit_data,
+	.dump_std_reg = rt1715_tcpc_dump_std_reg,
+	.alert_handler_cb = rt1715_tcpc_alert_handler_cb,
+	.get_status_register = rt1715_tcpc_get_status_register,
+	.clear_status_register = rt1715_tcpc_clear_status_register,
+	.mask_status_register = rt1715_tcpc_mask_status_register,
+	.set_drp_toggle = rt1715_tcpc_set_drp_toggle,
+	.get_chip_info = rt1715_tcpc_get_chip_info,
+	.set_low_power_mode = rt1715_tcpc_set_low_power_mode,
+	.sop_prime_enable = rt1715_tcpc_sop_prime_enable,
+	.set_bist_test_mode = rt1715_tcpc_set_bist_test_mode,
+	.set_alert_handler_cb = rt1715_tcpc_set_alert_handler_cb,
+};
+
+void rt1715_alert_cb(const struct device *port, struct gpio_callback *cb, gpio_port_pins_t pins)
+{
+	struct rt1715_data *data = CONTAINER_OF(cb, struct rt1715_data, alert_cb);
+
+	k_work_submit(&data->alert_work);
+}
+
+void rt1715_alert_work_cb(struct k_work *work)
+{
+	struct rt1715_data *data = CONTAINER_OF(work, struct rt1715_data, alert_work);
+	const struct device *dev = data->dev;
+	const struct rt1715_cfg *cfg = dev->config;
+	uint16_t alert_reg = 0;
+	uint16_t clear_flags = 0;
+
+	if (!data->initialized) {
+		return;
+	}
+
+	tcpci_tcpm_get_status_register(&cfg->bus, TCPC_ALERT_STATUS, &alert_reg);
+
+	while (alert_reg != 0) {
+		enum tcpc_alert alert_type = tcpci_alert_reg_to_enum(alert_reg);
+
+		if (alert_type == TCPC_ALERT_HARD_RESET_RECEIVED) {
+			LOG_DBG("hard rst received");
+			tcpci_init_alert_mask(dev);
+			data->cc_changed = true;
+		} else if (alert_type == TCPC_ALERT_FAULT_STATUS) {
+			uint8_t fault;
+
+			tcpci_tcpm_get_status_register(&cfg->bus, TCPC_FAULT_STATUS,
+						       (uint16_t *)&fault);
+			tcpci_tcpm_clear_status_register(&cfg->bus, TCPC_FAULT_STATUS,
+							 (uint16_t)fault);
+
+			LOG_DBG("fault: %02x", fault);
+		} else if (alert_type == TCPC_ALERT_EXTENDED_STATUS) {
+			uint8_t ext_status;
+
+			tcpci_tcpm_get_status_register(&cfg->bus, TCPC_EXTENDED_STATUS,
+						       (uint16_t *)&ext_status);
+			tcpci_tcpm_clear_status_register(&cfg->bus, TCPC_EXTENDED_STATUS,
+							 (uint16_t)ext_status);
+
+			data->cc_changed = true;
+			LOG_DBG("ext status: %02x", ext_status);
+		} else if (alert_type == TCPC_ALERT_POWER_STATUS) {
+			uint8_t pwr_status;
+
+			tcpci_tcpm_get_status_register(&cfg->bus, TCPC_POWER_STATUS,
+						       (uint16_t *)&pwr_status);
+			tcpci_tcpm_clear_status_register(&cfg->bus, TCPC_POWER_STATUS,
+							 (uint16_t)pwr_status);
+
+			LOG_DBG("power status: %02x", pwr_status);
+		} else if (alert_type == TCPC_ALERT_EXTENDED) {
+			uint8_t alert_status;
+
+			tcpci_tcpm_get_status_register(&cfg->bus, TCPC_EXTENDED_ALERT_STATUS,
+						       (uint16_t *)&alert_status);
+			tcpci_tcpm_clear_status_register(&cfg->bus, TCPC_EXTENDED_ALERT_STATUS,
+							 (uint16_t)alert_status);
+
+			LOG_DBG("ext alert: %02x", alert_status);
+		} else if (alert_type == TCPC_ALERT_MSG_STATUS) {
+			data->msg_pending = true;
+			rt1715_tcpc_rx_fifo_enqueue(dev);
```

Comment:
These should be done in the opposite order.  You also shouldn't set `msg_pending = true` if there is an error reading the receive message.

Reply:
fixed, thanks